byway/src/main/webapp/home.js

```diff
@@ -86,22 +92,49 @@ function showIncompleteTrip(tripNum, trip, isDestinationsMissing, isInterestsMis
  * @param {Number} tripNum 
  * @param {Trip} trip
  */
-function showCompleteTrip(tripNum, trip){
+async function showCompleteTrip(trip){
   let container = document.getElementById("past-trips-container");
   let pastTrip = document.createElement('div');
   pastTrip.className = "past-trip";
   let title = document.createElement('a');
+  title.id = "title-" + trip.keyString;
+  title.innerText = "Your Trip To:"
   pastTrip.append(title);
   let mapContainer = document.createElement('div');
   mapContainer.className = 'map';
-  mapContainer.id = "map-" + trip.keyString
+  mapContainer.id = "map-" + trip.keyString;
   pastTrip.append(mapContainer);
   container.append(pastTrip);
   initMap(trip.start, trip.start, trip.route, trip.keyString);
-  title.innerText = "Trip #" + tripNum;
+  await constructTripTitle(trip, trip.keyString);
   title.href = configureTripKeyForPath(trip.keyString, "/routepage.html");
 }
 
+
+/**
+ * Updates the title of specifice trip
+ * @param {Trip} trip
+ * @param {String} keyString trip's key as a string
+ */
+async function constructTripTitle(trip, keyString){
+  trip.destinations.forEach(async (destination) => {
```

Comment:
The array `forEach` function does not wait for promises to complete any more than `map` does. If you want to wait you'd have to use `for( ... of .. )` with await.

Reply:
fixed!